src/runtime/pkg/containerd-shim-v2/service.go

```diff
@@ -960,6 +960,11 @@ func (s *service) Shutdown(ctx context.Context, r *taskAPI.ShutdownRequest) (_ *
 
 	s.cancel()
 
+	// We must clean up the socket file before os.Exit(0)
+	if address, err := cdshim.ReadAddress("address"); err == nil {
+		_ = cdshim.RemoveSocket(address)
```

Comment:
I don't think we should halt shutdown if this fails, but I definitely think it'd be worth logging it

Reply:
Thanks for your suggestion, I have modified the code to print log when deletion fails